engine/pkg/rpcutil/server.go

```diff
@@ -88,6 +88,29 @@ func (l *LeaderClientWithLock[T]) Close() {
 	}
 }
 
+// rpcLimiter is a customed rate limiter, which delegates Allow of rate.Limiter,
+// and provides an allow list with a higher priority.
+type rpcLimiter struct {
+	limiter   *rate.Limiter
+	allowList []string
```

Comment:
Why not use a `map` here?

Reply:
In fact for small data size, slice search is faster than map search.
Given a sample benchmark: https://gist.github.com/amyangfei/aa9f3edb2785dafb86296b148da56c9d

```go
➜  go test -bench='Benchmark' .
goos: darwin
goarch: arm64
pkg: bench
BenchmarkSliceSearch-8           9815140               115.0 ns/op
BenchmarkMapSearch-8             7893643               151.5 ns/op
PASS
ok      bench   2.687s
```

Reply:
There are similar tests, https://darkcoding.net/software/go-slice-search-vs-map-lookup/
I think both slice and map are ok here, since it is not a hot path.